daphne/src/mock_leader_memory.rs

```diff
@@ -0,0 +1,264 @@
+// Copyright (c) 2022 Cloudflare, Inc. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+
+//! This module implements the necessary in memory datastructures necessary to implement an
+//! in-memory leader. For a real production implementation this should not be used as it means a
+//! machine crash or shutdown would cause jobs to be lost.
+
+use std::collections::{HashMap, VecDeque};
+
+use rand::{thread_rng, Rng};
+use url::Url;
+
+use crate::{
+    error::DapAbort,
+    fatal_error,
+    messages::{Base64Encode, BatchId, BatchSelector, Collection, CollectionJobId, Report, TaskId},
+    roles::leader::WorkItem,
+    DapAggregationParam, DapBatchBucket, DapCollectionJob, DapError, DapQueryConfig, DapTaskConfig,
+};
+
+#[derive(Default)]
+pub struct MockLeaderMemory {
```

Comment:
As discussed earlier today. Don't forget to rename the module as well. 
```suggestion
pub struct InMemoryLeaderState {
```